### PR TITLE
feat: set up Vitest

### LINKED ISSUE #40: Feature: add Vitest to the project
Add Vitest to the project, so that the starter kit has a testing framework in place.

Things to be done:
- Install [Vitest](https://vitest.dev/)
- For the vitest.config.ts file, see the [Astro docs for configuring Vitest](https://dev.to/opensauced/migrating-from-jest-to-vitest-for-your-react-application-1b75)
- Create one dummy test
- Add a step to run vitest in lint-staged for code files. See https://github.com/nickytonline/astro-partykit-starter/blob/c8fcfeb7624dd43a7259e7a221992decfbedf209/package.json#L42-L45
- Here's an example config for running it in lint-staged. https://github.com/open-sauced/app/blob/f9ea05308ec63d110a30cb8ada3b195fd3798ffc/package.json#L161

### PR DESCRIPTION
**Pull request by adiati98**
## Description

This PR adds [Vitest](https://vitest.dev/) to the project with details as follow:

- Install Vitest
- Add `vitest.config.ts` and the configuration.
- Add `test` to scripts in `package.json`.
- Add types for Vitest in `tsconfig.json`.
- Add a dummy test in `basics.tests.ts`.
- Update `lint-staged` with vitest.
- Add command to run Vitest to README.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] 📦 Chore
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #40 

## Steps to QA

1. Run `npm install`.
2. Go to `test` folder inside `src` folder.
3. Open `basic.tests.ts`.
4. Run `npm run test`.
5. You will see on the terminal that 2 tests in 1 file are passed.
6. Change a number in the 1st test or a word in the 2nd test.
7. You will see the information of which line of code is not correct and that there is 1 failed test in 1 file.

## Added to documentation?

- [x] 📜 README.md
- [ ] 📜 CONTRIBUTING.md
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by adiati98** [comment]
@nickytonline @jdwilkin4 

If you see the `force-pushed`, that was because I pushed an empty commit to try resolving `deploy/netlify Expected — Waiting for status to be reported` based on [this resource](https://stackoverflow.com/a/58190576). But it didn't resolved it.

But then I saw [this thread](https://github.com/orgs/community/discussions/26698) which probably worth to look into.

File: test/basic.tests.ts
Comment (nickytonline) [inline test/basic.tests.ts:21]: non-blocking: this test is just an example test, so we can keep it to a bare minimum.

```suggestion
test("is true", () => {
  expect(true).toBe(true);
});
```

File: vitest.config.ts
Comment (nickytonline) [inline vitest.config.ts:6]: Refactor (non-blocking): Most projects I've worked on [co-locate tests](https://kentcdodds.com/blog/colocation). I'd recommend we do that in this project as well, but I'll let others chime in.

**Review by nickytonline — APPROVED** [review]
Thanks for the PR @adiati98! This looks good to me. I left some comments, but it doesn't block the PR.

🚀

File: vitest.config.ts
Comment (adiati98) [reply vitest.config.ts:6]: I don't understand co-locate yet. I'll take a look at this tomorrow. Thanks, @nickytonline!

Comment (jdwilkin4) [reply vitest.config.ts:6]: Yeah, I like the idea of co located the tests. 

@adiati98 
Kent's article gives a good explanation for co locating tests. 
But also, here is an example of Nick doing this for the open sauced app repo
https://github.com/open-sauced/app/pull/2275/files

Comment (adiati98) [reply vitest.config.ts:6]: I took a look at the OpenSauced's app and kind of understand that we want to put the files that we want to test in the same folder as the tests files.

In this PR, do we want to rename the `test` to `lib/utils` folder and rename the file to be `.test.ts` ?

Comment (nickytonline) [reply vitest.config.ts:6]: Since this is just a "test" test file it's probably fine like this for now, but we can probably move it into the root of the `src` folder in this PR. Going forward, if tests need to be written, they should reside in the same folder as the code.

We can add that to the contributing guidelines here or in a follow up PR.

Comment (adiati98) [reply vitest.config.ts]: I've moved the `test` folder to the root of the `src` folder.

And I've updated the README to add the test folder into the project structure. Please let me know if this what you meant by adding to the guidelines.

Thanks, @nickytonline!

**Review by nickytonline — DISMISSED** [review]
Thanks for making the changes @adiati98!

File: package.json
Comment (nickytonline) [inline package.json:8]: This doesn’t have to be for this PR. It can be a small follow up PR, but it’d be good to have a `prebuild` script that runs `vitests run`, so if there is a test failure the build will fail.

**Comment by nickytonline** [comment]
This just needs another review/approval.

**Comment by nickytonline** [comment]
This just needs another approval.

**Comment by adiati98** [comment]
> This just needs another approval.

I can't approve my own PR. 😅
If you have authority to merge it in, that'd be great, @nickytonline 😊

**Comment by nickytonline** [comment]
I'll go ahead and do that. I'll also change the policy to only require one approval since it's just us maintaining the project now. 😅

**Review by nickytonline — APPROVED** [review]
Thanks for this @adiati98! 🚢

**Pull request merged**
